lib/client/api.go

```diff
@@ -4974,6 +4974,37 @@ func ParseDynamicPortForwardSpec(spec []string) (DynamicForwardedPorts, error) {
 	return result, nil
 }
 
+// PortMapping represents a mapping of LocalPort to TargetPort, e.g., "1337:42".
+type PortMapping struct {
+	LocalPort  int
+	TargetPort int
+}
+
+// ParsePortMapping parses textual form of port mapping (e.g., "1337:42") into a struct. It accepts
+// a single port number as well (e.g., "42"). Both numbers must be between 0 and 65535.
+func ParsePortMapping(rawPorts string) (PortMapping, error) {
+	if rawPorts == "" {
+		return PortMapping{}, nil
+	}
+
+	parts := strings.SplitN(rawPorts, ":", 2)
```

Comment:
`strings.Cut` is almost always better than `strings.SplitN(2)` IMO.

Reply:
Oh TIL, I didn't know about it. With `strings.SplitN`, I always forget if `SplitN(1)` or `SplitN(2)` does the equivalent of `strings.Cut`.